src/dbcp/commands/publish.py

```diff
@@ -251,14 +285,23 @@ def publish_outputs(
     blob.upload_from_string(metadata.to_yaml())
     logger.info(f"Uploaded metadata to gs://{bucket_name}/{destination_blob_name}")
 
-    if upload_to_big_query:
-        if target:
-            for directory in directories:
+    if target is not None:
+        for directory in directories:
+            if upload_to_big_query:
                 load_parquet_files_to_bigquery(
                     output_bucket, directory, metadata.version, target
                 )
-        else:
-            logger.warning("No target schema provided. Skipping BigQuery upload.")
+            # At this point postgres is only used for production data mart tables
+            if (
+                upload_to_postgres
+                and ("data_mart" in str(directory))
+                and (target == "prod")
+            ):
```

Comment:
With this configuration we could in theory update prod tables from a branch (not `main`) and it would still upload to postgres, which I think is fine.